components/director/internal/tenantfetcher/service.go

```diff
@@ -59,11 +71,24 @@ func (s Service) SyncTenants() error {
 	if err != nil {
 		return err
 	}
+	tenantsToCreate = s.dedupeTenants(tenantsToCreate)
+
 	tenantsToDelete, err := s.getTenantsToDelete()
 	if err != nil {
 		return err
 	}
 
+	deleteTenantsMap := make(map[string]model.BusinessTenantMappingInput)
+	for _, ct := range tenantsToDelete {
+		deleteTenantsMap[ct.ExternalTenant] = ct
+	}
+
+	for i := len(tenantsToCreate) - 1; i >= 0; i-- {
+		if _, found := deleteTenantsMap[tenantsToCreate[i].ExternalTenant]; found {
+			tenantsToCreate = append(tenantsToCreate[:i], tenantsToCreate[i+1:]...)
```

Comment:
Is it possible to have a delete event before a create event which would mean that we would still want to have the tenant created (kept in the `tenantsToCreate` map)?

Reply:
yes, this is possible. There is also an issue in github on this. This will be fixed in a future improvement as it has not happened in prod and it is not expected to have such situation. Of course `expected` is not something to rely on.